File: WatchTower.cpp
#include "WatchTower.hpp"

void
WatchTower::preturnAction() {
	for (Player *e : this->owner->getEnemies()) {
		for (Unit *u : e->getUnits())
			if (this->distance(u) <= this->attack_range)
				AttackMapItem::attack(u->x, u->y);

		for (Building *b : e->getUnits())
			if (this->distance(b) <= this->attack_range)
				AttackMapItem::attack(b->x, b->y);
	}
}

File: ruby/WatchTower.rb
require './Building.rb'
# not ruby yet!


class WatchTower < Building
public:
	WatchTower(std::string popis, Player owner) : Building(popis, owner) {}
	WatchTower(Player owner) : Building("WatchTower", owner) {
		@type = bt_WatchTower
		@hitpoints = 100
		@attack_range = 2
		@attack_damage = 1
		owner.addScore(50)
	}

	void preturnAction() { self.attack(); }

	static Building create(Player owner) { return WatchTower.new(owner); }
end

